modules_core/org.openbravo.client.application/src/org/openbravo/client/application/process/BaseProcessActionHandler.java

```diff
@@ -169,12 +210,124 @@ protected final JSONObject execute(Map<String, Object> parameters, String conten
     }
   }
 
+  /**
+   * Updates the result JSONObject by setting the REFRESH_PARENT property.
+   * If the property is not present or is set to true, the method ensures
+   * that the REFRESH_PARENT flag is true, meaning the parent needs to be refreshed.
+   *
+   * @param result the JSONObject containing the process results.
+   * @return the updated JSONObject with the REFRESH_PARENT flag set.
+   * @throws JSONException if there is an error manipulating the JSON object.
+   */
   private JSONObject doRefreshParent(JSONObject result) throws JSONException {
     boolean deRefresh = !result.has(REFRESH_PARENT) || result.getBoolean(REFRESH_PARENT);
     result.put(REFRESH_PARENT, deRefresh);
     return result;
   }
 
+  /**
+   * Checks if the file size provided in the file parameters is within the allowed limit.
+   * It retrieves the maximum allowed file size from the system preferences and compares it
+   * to the file size passed in the fileParam map.
+   *
+   * @param fileParam a map containing information about the file, including its size.
+   * @return true if the file size is within the allowed limit, false otherwise.
+   */
+  private boolean isFileSizeWithinLimit(Map<String, Object> fileParam) {
+    String maximumFileSize = this.getMaximumUploadedFileSize();
+    if (maximumFileSize != null) {
+      long fileSizeMB = (long) fileParam.get(PARAM_FILE_SIZE) / 1000000;
```

Comment:
1000000???

Reply:
check org.apache.commons.lang3.math.NumberUtils util class for Long methods it's like as StringUtils.